File: src/test/java/kitchenpos/domain/FakeOrderTableRepository.java
package kitchenpos.domain;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public class FakeOrderTableRepository implements OrderTableRepository {

    private Map<UUID, OrderTable> inMemory = new HashMap<>();

    @Override
    public OrderTable save(OrderTable orderTable) {
        inMemory.put(orderTable.getId(), orderTable);
        return orderTable;
    }

    @Override
    public Optional<OrderTable> findById(UUID orderTableId) {
        return Optional.ofNullable(inMemory.get(orderTableId));
    }

    @Override
    public List<OrderTable> findAll() {
        return inMemory.values().stream().toList();
    }
}

File: src/test/java/kitchenpos/domain/FakeOrderTableRepository.kt
package kitchenpos.domain

import java.util.Optional
import java.util.UUID

class FakeOrderTableRepository : InMemoryRepository<OrderTable>(), OrderTableRepository {
    override fun findById(id: UUID): Optional<OrderTable> {
        return items.find { it.id == id }?.let { Optional.of(it) } ?: Optional.empty()
    }
}
